internal/sidecarlogresults/sidecarlogresults_test.go

```diff
@@ -0,0 +1,105 @@
+package sidecarlogresults
+
+import (
+	"bytes"
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/pipeline/test/diff"
+)
+
+func TestResultEncode(t *testing.T) {
+	for _, c := range []struct {
+		desc   string
+		result SidecarLogResult
+	}{{
+		desc: "good result",
+		result: SidecarLogResult{
+			Name:  "foo",
+			Value: "bar",
+		},
+	}} {
+		t.Run(c.desc, func(t *testing.T) {
+			got := new(bytes.Buffer)
+			encode(got, &c.result)
+			want, err := json.Marshal(&c.result)
+			if err != nil {
+				t.Error(err)
+			}
+			// encode adds a newline character at the end.
+			// We need to do the same before comparing
+			want = append(want, []byte("\n")...)
+			if d := cmp.Diff(want, []byte(got.String())); d != "" {
```

Comment:
There's also https://pkg.go.dev/bytes#Buffer.Bytes, but I'm curious if the output would be easier to read as a byte buffer or a string? 🤔